tests/tests/tier0/bluechi-is-online-agent-wait/test_bluechi_is_online_agent_wait.py

```diff
@@ -0,0 +1,140 @@
+MAX_WAIT_PARAM_VALUE = int(os.getenv("MAX_WAIT_PARAM_VALUE", 20000))
+SLEEP_DURATION = int(os.getenv("SLEEP_DURATION", 2))
+
+
+class ResultFuture:
+    def __init__(self):
+        self.result = None
+        self.output = ""
+
+
+def check_agent(
+    bluechi_is_online: BluechiIsOnline,
+    future: ResultFuture,
+    description: str,
+    wait_time: int = 0,
+    expected_result: int = 0,
+):
+    result = bluechi_is_online.agent_is_online(
+        wait_time,
+    )
+    future.result = result
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    node_foo = nodes[NODE_FOO]
+
+    # Test 1: Start agent and verify --wait option returns 0 immediately
+    with Timeout(
+        IMMEDIATE_RETURN_TIMEOUT, "bluechi-is-online didn't return immediately"
+    ):
+        LOGGER.debug("Starting test number 1 - agent should be online.")
+        assert node_foo.bluechi_is_online.agent_is_online(wait=IMMEDIATE_RETURN_TIMEOUT)
+
+    # Test 2: Stop agent and verify --wait returns 1 after wait time is expired
+    node_foo.systemctl.stop_unit("bluechi-agent")
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+    LOGGER.debug("Starting test number 2 - agent should remain offline.")
+
+    start_time = time.time()
+    # extend the test timeout here a bit since we are wrapping bluechi-is-online
+    with Timeout(
+        WAIT_PARAM_VALUE + 1, "bluechi-is-online did not exit within the specified time"
+    ):
+        result = node_foo.bluechi_is_online.agent_is_online(WAIT_PARAM_VALUE)
+
+    assert (
+        not result
+    ), f"Expected bluechi-is-online with --wait={WAIT_PARAM_VALUE} on node {NODE_FOO} to exit false"
+    assert (
+        start_time - time.time() <= WAIT_PARAM_VALUE + 1
+    ), "Expected around 5 second for bluechi-is-online to exit"
+
+    # Test 3: Stop agent, run 'bluechi-is-online agent --wait', start agent and verify --wait returns 0 before the
+    # wait time expires
+    try:
+        with Timeout(MAX_WAIT_PARAM_VALUE, "Timeout during Test 3"):
+            LOGGER.debug(
+                "Starting test number 3, stopping agent to ensure it is inactive before starting `bluechi-is-online`."
+            )
+            node_foo.systemctl.stop_unit("bluechi-agent")
+            assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+            LOGGER.debug("Agent confirmed inactive after stopping.")
+
+            result_future_wait = ResultFuture()
+            start_time = time.time()
+            LOGGER.debug(
+                "Starting `bluechi-is-online` thread with wait time of 20 seconds."
+            )
+            check_thread_wait = threading.Thread(
+                target=check_agent,
+                args=(
+                    node_foo.bluechi_is_online,
+                    result_future_wait,
+                    "Checking if agent comes online",
+                    MAX_WAIT_PARAM_VALUE,
```

Comment:
Why do we want to pass that big timeout to `wait` parameter? Isn't those 5 secs in `WAIT_PARAM_VALUE` enough?

Reply:
Done. Changed to 5 seconds (WAIT_PARAM_VALUE)